sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -915,6 +915,52 @@ __ESIMD_API void lsc_block_store(T *p, __ESIMD_NS::simd<T, NElts> vals) {
                                               vals.data());
 }
 
+namespace detail {
+// Compile-time checks for lsc_load2d/store2d restrictions.
+template <typename T, int BlockWidth, int BlockHeight, int NBlocks,
+          bool Transposed, bool Transformed, bool IsStore = false>
+constexpr void check_lsc_block_2d_restrictions() {
+  constexpr int GRFByteSize = BlockWidth * BlockHeight * NBlocks * sizeof(T);
+  static_assert(!IsStore || GRFByteSize <= 512,
+                "2D store supports 512 bytes max");
+  static_assert(IsStore || GRFByteSize <= 2048,
+                "2D load supports 2048 bytes max");
+  static_assert(!Transposed || !Transformed,
+                "Transposed and transformed is not supported");
+  static_assert(!Transposed || (Transposed && NBlocks == 1),
```

Comment:
This line is equivalent to the shorter version:
```suggestion
  static_assert(!Transposed || NBlocks == 1,
```
The checks verifying _only one_ of {Transposed,Transformed} would be more readable being placed under those 'if constexpr (Transposed)' below at L936 and 'if constexpr (Transformed)' at L943